src/adc.rs

```diff
@@ -547,6 +554,56 @@ impl<C> From<C> for Channels
     }
 }
 
+/// Helper trait to convert something into a channels value.  This is
+/// necessary, because we need to have a channels value to set up the
+/// ADC scanning, but also we need to "keep" the original value in the
+/// ADC until the conversion completes
+pub trait IntoChannels {
+    fn into_channels(&self) -> Channels;
+}
+
+impl IntoChannels for Channels {
+    fn into_channels(&self) -> Channels {
+	self.clone()
+    }
+}
+
+impl<C> IntoChannels for C
+    where C: Channel<Adc<Ready>, ID=u8>
+{
+    fn into_channels(&self) -> Channels {
+        Channels{flags: 0x1 << C::channel()}
```

Comment:
Might be nicer to re-use `add` here. The compiler should be able to optimize that (if it doesn't, it might be possible to convince it by adding `#[inline(always)]` to `add`).